iceberg/openhouse/internalcatalog/src/main/java/com/linkedin/openhouse/internal/catalog/OpenHouseInternalCatalog.java

```diff
@@ -128,7 +132,22 @@ public boolean dropTable(TableIdentifier identifier, boolean purge) {
 
   @Override
   public void renameTable(TableIdentifier from, TableIdentifier to) {
-    throw new UnsupportedOperationException("Rename Tables not implemented yet");
+    TableIdentifier fromTableId = TableIdentifier.of(from.namespace().toString(), from.name());
+    Table fromTable = loadTable(fromTableId);
+    String tableClusterId = fromTable.properties().get(CatalogConstants.OPENHOUSE_CLUSTERID_KEY);
+    TableUri tableUri =
+        TableUri.builder()
+            .clusterId(tableClusterId)
+            .databaseId(to.namespace().toString())
+            .tableId(to.name())
+            .build();
+    Transaction transaction = fromTable.newTransaction();
+    UpdateProperties updateProperties = transaction.updateProperties();
+    updateProperties.set(CatalogConstants.OPENHOUSE_TABLEID_KEY, to.name());
+    updateProperties.set(CatalogConstants.OPENHOUSE_DATABASEID_KEY, to.namespace().toString());
+    updateProperties.set(CatalogConstants.OPENHOUSE_TABLEURI_KEY, tableUri.toString());
+    updateProperties.commit();
+    transaction.commitTransaction();
```

Comment:
Can we add log here mentioning table properties updated due to rename and emit metrics as well?

Reply:
Which metric should we add? In the caller class `OpenHouseInternalRepository`, it is already recording metrics `REPO_TABLE_RENAME_TIME`, is another metric necessary?